src/main/java/com/minecolonies/coremod/network/messages/server/colony/OpenInventoryMessage.java

```diff
@@ -175,30 +146,17 @@ private void doCitizenInventory(final ServerPlayerEntity player)
         }
     }
 
-    private void doHutInventory(final ServerPlayerEntity player)
+    private void doHutInventory(final ServerPlayerEntity player, final IColony colony)
     {
-        if (checkPermissions(IColonyManager.getInstance().getClosestColony(player.getEntityWorld(), tePos), player))
-        {
-            @NotNull final TileEntityRack chest = (TileEntityRack) BlockPosUtil.getTileEntity(player.world, tePos);
+        @NotNull final TileEntityRack chest = (TileEntityRack) BlockPosUtil.getTileEntity(player.world, tePos);
 
-            NetworkHooks.openGui(player, chest, packetBuffer -> packetBuffer.writeVarInt(colonyId).writeBlockPos(chest.getPos()));
-        }
+        NetworkHooks.openGui(player, chest, packetBuffer -> packetBuffer.writeVarInt(colony.getID()).writeBlockPos(chest.getPos()));
     }
 
     private void doFieldInventory(final ServerPlayerEntity player)
     {
-        if (checkPermissions(IColonyManager.getInstance().getClosestColony(player.getEntityWorld(), tePos), player))
-        {
-            @NotNull final ScarecrowTileEntity scarecrowTileEntity = (ScarecrowTileEntity) BlockPosUtil.getTileEntity(CompatibilityUtils.getWorldFromEntity(player), tePos);
-            NetworkHooks.openGui(player, scarecrowTileEntity);
-
-        }
-    }
-
-    private static boolean checkPermissions(final IColony colony, final ServerPlayerEntity player)
-    {
-        //Verify player has permission to change this huts settings
-        return colony.getPermissions().hasPermission(player, Action.MANAGE_HUTS);
```

Comment:
does this mean the permission doesn't get checked at all in this packet?

Reply:
look at the base class: `AbstractColonyServerMessage`

Reply:
that's a major part of the boiler plate i removed

Reply:
Seems ok then.
I don't know too much about minecraft/minecolonies networking, so i just looked for changes in logic that seemed important.